glean-core/preview/src/core_metrics.rs

```diff
@@ -0,0 +1,113 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use glean_core::{metrics::StringMetric, CommonMetricData, Lifetime};
+
+/// Metrics included in every ping as `client_info`.
+#[derive(Debug)]
+pub struct ClientInfoMetrics {
+    /// The build identifier generated by the CI system (e.g. "1234/A").
+    pub app_build: String,
+    /// The user visible version string (e.g. "1.0.3").
+    pub app_display_version: String,
+    /// The channel the application is being distributed on.
+    pub app_channel: Option<String>,
+    /// The name of the operating system.
+    pub os: String,
+    /// The user-visible version of the operating system (e.g. "1.2.3").
+    pub os_version: String,
+    /// The manufacturer of the device the application is running on.
+    pub device_manufacturer: String,
```

Comment:
Does this and the following even apply to Desktop?

Reply:
The information exists and you [can get it](https://github.com/mozilla/glean/blob/master/glean-core/python/glean/hardware.py).  Whether we want to is a separate question.

Reply:
> 
> 
> The information exists and you [can get it](https://github.com/mozilla/glean/blob/master/glean-core/python/glean/hardware.py). Whether we want to is a separate question.

Ah right! I'd be for stubbing out these values and add proper detection out of the prototype phase.

Reply:
Another question is whether this information is meaningful on Desktop (where, e.g., I could add HW components to wildly change how my manufactured model works compared to others).

Reply:
Oh yes, the data of course does exist.
But what chutten said and also Firefox might have the right ways to detect this (or similar information) already, which is why I wouldn't want to have yet-another-way to read this out.

Unfortunately `device_manufacturer` _is_ a required field in the schema (to my knowledge), which is why I added it in the first place.

Reply:
> 
> Unfortunately `device_manufacturer` _is_ a required field in the schema (to my knowledge), which is why I added it in the first place.

Since this is a prototype ping, we could just stub it out as we did for our other POCs, if @chutten agrees.